src/sync/regular.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * @module sync
+ */
+
+import {hashTreeRoot} from "@chainsafe/ssz";
+
+import {BeaconBlock, Attestation} from "../types";
+import {BLOCK_TOPIC, ATTESTATION_TOPIC} from "../constants";
+import {IBeaconDb} from "../db";
+import {IBeaconChain} from "../chain";
+import {INetwork} from "../network";
+import {OpPool} from "../opPool";
+import {ILogger} from "../logger";
+
+import {slotToEpoch} from "../chain/stateTransition/util";
+
+export class RegularSync {
+  private db: IBeaconDb;
+  private chain: IBeaconChain;
+  private network: INetwork;
+  private opPool: OpPool;
+  private logger: ILogger;
+
+  public constructor(opts, {db, chain, network, opPool, logger}) {
+    this.db = db;
+    this.chain = chain;
+    this.network = network;
+    this.opPool = opPool;
+    this.logger = logger;
+  }
+
+  public async receiveBlock(block: BeaconBlock): Promise<void> {
+    // TODO: skip block if its a known bad block
```

Comment:
how would you know if it's a bad block, `process_block` should drop it I guess

Reply:
I'm thinking we store bad block hashes in the db. And add a code path to `chain.receiveBlock` that does the db call.